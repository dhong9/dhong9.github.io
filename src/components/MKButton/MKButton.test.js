import renderer from "react-test-renderer";
import MKButtonRoot from "components/MKButton/MKButtonRoot";

// Mocks
jest.mock("components/MKButton", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <button>Mock Button</button>)
    };
});

jest.mock("@mui/material/styles", () => ({
    ...jest.requireActual("@mui/material/styles"),
    styled: jest.fn().mockReturnValue(jest.fn().mockReturnValue("mock-style")),
  }));

describe("MKButton", () => {
    it("renders", () => {
        const component = renderer.create(<MKButtonRoot />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders gradient", () => {
        const component = renderer.create(<MKButtonRoot variant="gradient" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});