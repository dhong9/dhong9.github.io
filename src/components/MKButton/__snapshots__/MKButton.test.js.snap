// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKAlert renders 1`] = `<mock-style />`;

exports[`MKAlert renders gradient 1`] = `
<mock-style
  variant="gradient"
/>
`;
